src/app/src/components/WelcomeModal.js

```diff
@@ -1,31 +1,61 @@
-import {
-    Button,
-    Modal,
-    ModalOverlay,
-    ModalContent,
-    ModalBody,
-    ModalHeader,
-    Heading,
-    Text,
-} from '@chakra-ui/react';
+import { useState } from 'react';
+import { Button, Modal, ModalOverlay } from '@chakra-ui/react';
+
+import HowItWorks from './ModalSections/HowItWorks';
+import Introduction from './ModalSections/Introduction';
+import FileUpload from './ModalSections/FileUpload';
+
+const sectionMap = {
+    0: Introduction,
+    1: HowItWorks,
+    2: FileUpload,
+};
+const firstSection = 0;
+const lastSection = Object.keys(sectionMap).length - 1;
 
 export default function WelcomeModal() {
+    const [currentSection, setCurrentSection] = useState(0);
+
+    const goToNextSection = () => {
+        setCurrentSection(previousSection =>
+            Math.min(previousSection + 1, lastSection)
+        );
+    };
+
+    const goToPreviousSection = () => {
+        setCurrentSection(previousSection =>
+            Math.max(previousSection - 1, firstSection)
+        );
+    };
+
+    const PreviousButton = (
+        <Button onClick={goToPreviousSection}>Previous</Button>
+    );
+
+    const ContinueButton = (
+        <Button
+            bg='gray.900'
+            color='white'
+            boxShadow='base'
+            onClick={goToNextSection}
+        >
+            Continue
+        </Button>
+    );
```

Comment:
I would recommend we rely on variants:

```suggestion
    const PreviousButton = (
        <Button variant='secondary' onClick={goToPreviousSection}>
            Previous
        </Button>
    );

    const ContinueButton = (
        <Button variant='cta' onClick={goToNextSection}>
            Continue
        </Button>
    );
```

And define them as done in Figma here: https://www.figma.com/file/tJPZ0lkT5C3SAyS2g2R1aA/IoW---Boundary-Sync-Tool?node-id=234%3A12160

```js
const Button = {
    baseStyle: {
        fontFamily: `'Inter', sans-serif`,
        borderRadius: '4px',
        fontWeight: 600,
    },
    variants: {
        cta: {
            bg: 'gray.900',
            color: 'white',
            _hover: {
                bg: 'gray.600',
                textDecoration: 'underline',
            },
            _disabled: {
                bg: 'gray.400',
            },
        },
        secondary: {
            bg: 'white',
            borderWidth: '1px',
            borderColor: 'gray.200',
            color: 'gray.600',
            _hover: {
                bg: 'gray.200',
                textDecoration: 'underline',
            },
            _disabled: {
                color: 'gray.700',
            },
        },
    },
};
```

Reply:
Good idea! 57e6ea542408a3c9dab0d1ceaf9547987fdef2e8